plugins/locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/CompassManager.java

```diff
@@ -0,0 +1,74 @@
+package com.mapbox.mapboxsdk.plugins.locationlayer;
+
+import android.content.Context;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+import android.os.SystemClock;
+
+/**
+ * This manager class handles compass events such as starting the tracking of device bearing, or when a new compass
+ * update occurs.
+ *
+ * @since 0.1.0
+ */
+class CompassManager implements SensorEventListener {
+
+  private final SensorManager sensorManager;
+  private CompassListener compassListener;
+
+  private Sensor rotationVectorSensor;
+  private float[] matrix = new float[9];
+  private float[] orientation = new float[3];
+
+  // CompassManager data
+  private long compassUpdateNextTimestamp = 0;
+
+  CompassManager(Context context, CompassListener compassListener) {
+    sensorManager = (SensorManager) context.getSystemService(Context.SENSOR_SERVICE);
+    rotationVectorSensor = sensorManager.getDefaultSensor(Sensor.TYPE_ROTATION_VECTOR);
+
+    this.compassListener = compassListener;
+  }
+
+  void onStart() {
+    sensorManager.registerListener(this, rotationVectorSensor, SensorManager.SENSOR_DELAY_GAME);
+  }
+
+  void onStop() {
+    sensorManager.unregisterListener(this, rotationVectorSensor);
+  }
+
+  boolean isSensorAvailable() {
+    return rotationVectorSensor != null;
+  }
+
+  @Override
+  public void onSensorChanged(SensorEvent event) {
+
+    // check when the last time the compass was updated, return if too soon.
+    long currentTime = SystemClock.elapsedRealtime();
+    if (currentTime < compassUpdateNextTimestamp) {
+      return;
+    }
+
+    if (event.sensor.getType() == Sensor.TYPE_ROTATION_VECTOR) {
```

Comment:
Should the fallback be the same way we used to get bearing measurements before the `TYPE_ROTATION_VECTOR` was changed? https://github.com/mapbox/mapbox-gl-native/blob/5870fceb25fe7f9c67b9275236d2bab2339384f3/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/widgets/MyLocationView.java#L548-L556